operator/v1/types.go

```diff
@@ -354,6 +359,22 @@ type NodePlacement struct {
 	Tolerations []v1.Toleration `json:"tolerations,omitempty"`
 }
 
+// HubApiServerHostAlias holds the mapping between IP and hostnames that will be injected as an entry in the
+// pod's hosts file.
+type HubApiServerHostAlias struct {
+	// IP address of the host file entry.
+	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=`^((0|[1-9]\d|1\d\d|2[0-4]\d|25[0-5])\.){3}(0|[1-9]\d|1\d\d|2[0-4]\d|25[0-5])$`
+	IP string `json:"ip,omitempty" protobuf:"bytes,1,opt,name=ip"`
```

Comment:
no need to add protobuf, it is not supported in crd.

Reply:
Done.   And i  think required field do not need omitempty tag.